### PR TITLE
feat(components): [el-pagination] add `teleported` prop

### LINKED ISSUE #14070: [Component] [pagination] toplayer全屏模式下分页下拉框无法显示
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.0`
- Element Plus Version: `2.3.9`
- Browser / OS: `chrome 116.0.5845.110`
- Build Tool: `Vite`

### Related Component

- `el-pagination`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6IjxzY3JpcHQgc2V0dXAgbGFuZz1cInRzXCI+XG5pbXBvcnQgeyByZWYsIHZlcnNpb24gYXMgdnVlVmVyc2lvbiB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IHZlcnNpb24gYXMgZXBWZXJzaW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuaW1wb3J0IHsgRWxlbWVudFBsdXMgfSBmcm9tICdAZWxlbWVudC1wbHVzL2ljb25zLXZ1ZSdcblxuXG5jb25zdCBtc2cgPSByZWYoJ0hlbGxvIFdvcmxkIScpXG5jb25zdCBwYWdlc2l6ZSA9IHJlZigxMClcbmNvbnN0IGN1cnJlbnRQYWdlID0gcmVmKDEpXG5jb25zdCBmdWxsc2NyZWVuID0gKGVsZW1lbnQpPT4ge1xuICBpZiAoZG9jdW1lbnQubW96RnVsbFNjcmVlbkVuYWJsZWQpIHtcbiAgICByZXR1cm4gUHJvbWlzZS5yZWplY3QobmV3IEVycm9yKFwi5YWo5bGP5qih5byP6KKr56aB55SoXCIpKTtcbiAgfVxuICBsZXQgcmVzdWx0ID0gbnVsbDtcbiAgaWYgKGVsZW1lbnQucmVxdWVzdEZ1bGxzY3JlZW4pIHtcbiAgICByZXN1bHQgPSBlbGVtZW50LnJlcXVlc3RGdWxsc2NyZWVuKCk7XG4gIH0gZWxzZSBpZiAoZWxlbWVudC5tb3pSZXF1ZXN0RnVsbFNjcmVlbikge1xuICAgIHJlc3VsdCA9IGVsZW1lbnQubW96UmVxdWVzdEZ1bGxTY3JlZW4oKTtcbiAgfSBlbHNlIGlmIChlbGVtZW50Lm1zUmVxdWVzdEZ1bGxzY3JlZW4pIHtcbiAgICByZXN1bHQgPSBlbGVtZW50Lm1zUmVxdWVzdEZ1bGxzY3JlZW4oKTtcbiAgfSBlbHNlIGlmIChlbGVtZW50LndlYmtpdFJlcXVlc3RGdWxsc2NyZWVuKSB7XG4gICAgcmVzdWx0ID0gZWxlbWVudC53ZWJraXRSZXF1ZXN0RnVsbFNjcmVlbigpO1xuICB9XG4gIHJldHVybiByZXN1bHQgfHwgUHJvbWlzZS5yZWplY3QobmV3IEVycm9yKFwi5LiN5pSv5oyB5YWo5bGPXCIpKTtcbn1cbmNvbnN0IGRvbSA9IHJlZignZG9tJylcbmNvbnN0IHRvZ2dsZUZ1bGxzY3JlZW4gPSAoKT0+e1xuICBmdWxsc2NyZWVuKGRvbS52YWx1ZSlcbn1cbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxoMT57eyBtc2cgfX08L2gxPlxuIFxuICA8ZGl2IHJlZj1cImRvbVwiPlxuICAgIDxlbC1idXR0b24gdHlwZT0ncHJpbWFyeScgQGNsaWNrPVwidG9nZ2xlRnVsbHNjcmVlblwiPiBmdWxsc2NyZWVuIDwvZWwtYnV0dG9uPlxuICAgIDxlbC1pbnB1dCB2LW1vZGVsPVwibXNnXCIgLz5cbiAgICA8ZWwtcGFnaW5hdGlvblxuICAgICAgYmFja2dyb3VuZFxuICAgICAgOnBhZ2Utc2l6ZXM9XCJbMTAsIDIwLCA1MCwgODAsIDEwMF1cIlxuICAgICAgdi1tb2RlbDpwYWdlLXNpemU9XCJwYWdlc2l6ZVwiXG4gICAgICB2LW1vZGVsOmN1cnJlbnQtcGFnZT1cImN1cnJlbnRQYWdlXCJcbiAgICAgIGxheW91dD1cInRvdGFsLCBwcmV2LCBwYWdlciwgbmV4dCwgc2l6ZXMsIGp1bXBlclwiXG4gICAgICA6dG90YWw9XCIyMFwiXG4gICAgLz5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge31cbn0iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWUsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiX28iOnt9fQ==)

### Steps to reproduce

1. 点击fullscreen按钮
2. 点击分页size选择器

### What is Expected?

显示size下拉框选项

### What is actually happening?

不显示size下拉框选项

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by boomboy4**
分页组件新增teleported参数 默认为true  便于用户控制size下拉框是否插入body

closed #14070

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### <samp>🤖 Generated by Copilot at 4a7e171</samp>

Added a `teleported` prop to the `ElPagination` and `ElPaginationSizes` components to enable or disable teleporting the pagination size component to the body element. This improves the flexibility and usability of the pagination component.

## Related Issue

Fixes #\_\_\_.

### <samp>🤖 Generated by Copilot at 4a7e171</samp>

*  Add a new prop `teleported` to control the teleportation of the pagination size component ([link](https://github.com/element-plus/element-plus/pull/14072/files?diff=unified&w=0#diff-aa9b95a7d5e14b75b9b0e648d113658eed0640c0af056f7a98ae7a2233204141R19), [link](https://github.com/element-plus/element-plus/pull/14072/files?diff=unified&w=0#diff-32614fd50766d56c282c787fc9cff64c1c38af941ad44b8e351a453c228d8d3fR141-R147))
*  Pass the `teleported` prop from the `ElPagination` component to the `ElPaginationSizes` component in the template and the script ([link](https://github.com/element-plus/element-plus/pull/14072/files?diff=unified&w=0#diff-3f7beae15e5e16681edca143621d01ca0328e9d0487c4e647f8c7f2ed0b693a2R8), [link](https://github.com/element-plus/element-plus/pull/14072/files?diff=unified&w=0#diff-32614fd50766d56c282c787fc9cff64c1c38af941ad44b8e351a453c228d8d3fR379))
*  Set the default value of the `teleported` prop to `true` for the `ElPagination` component ([link](https://github.com/element-plus/element-plus/pull/14072/files?diff=unified&w=0#diff-32614fd50766d56c282c787fc9cff64c1c38af941ad44b8e351a453c228d8d3fR141-R147))

### CODE REVIEW

**Comment by boomboy4** [comment]
这个单元测试错误好像跟我提交的代码无关，有谁指导下该怎么处理吗

File: packages/components/pagination/src/pagination.ts [hunk 1]
```diff
@@ -137,6 +137,10 @@ export const paginationProps = buildProps({
     type: iconPropType,
     default: () => ArrowRight,
   },
+  /**
+   * @description whether Pagination size is teleported to body
+   */
+  teleported: Boolean,
```
Comment (btea) [inline packages/components/pagination/src/pagination.ts:143]: The default value written in the document is `true`, and the default value `true` should also be added here.

Comment (boomboy4) [reply packages/components/pagination/src/pagination.ts:143]: Yes, i can change it, but isn't teleported already defaulted to true on select?

Comment (btea) [reply packages/components/pagination/src/pagination.ts]: Yes, in the `select` component, there is a `teleported` attribute with a default value of `true`.

But after you modify the code, the `select` component here receives the value passed from the `pagination` component. If the `teleported` attribute in the `pagination` component does not give a default value, Vue will convert its default value to `false`. At this time, the value received by the select component becomes `false`.

Comment (boomboy4) [reply packages/components/pagination/src/pagination.ts]: Got it. Thanks.

**Review by btea — APPROVED** [review]
Thanks!

File: docs/en-US/component/pagination.md [hunk 2]
```diff
@@ -84,6 +84,7 @@ pagination/more-elements
 | next-text                           | text for the next button                                                                                                        | ^[string]                                                                         | ''                                   |
 | next-icon                           | icon for the next button, has a lower priority than `next-text`                                                                 | ^[string] / ^[Component]                                                          | ArrowRight                           |
 | disabled                            | whether Pagination is disabled                                                                                                  | ^[boolean]                                                                        | false                                |
+| teleported                          | whether Pagination select dropdown is teleported to the body                                                                    | ^[boolean]                                                                        | true                                |
```
Comment (tolking) [inline docs/en-US/component/pagination.md:87]:
```suggestion
| teleported ^(2.3.13)       | whether Pagination select dropdown is teleported to the body                                                                    | ^[boolean]                                                                        | true                                |
```

**Review by tolking — APPROVED** [review]
Thank for your contribution

**Pull request merged**